openedx/features/course_experience/tests/views/test_course_outline.py

```diff
@@ -350,7 +363,62 @@ def test_resume_course(self):
         self.assertContains(response, 'Resume Course', count=2)
 
         content = pq(response.content)
-        self.assertTrue(content('.action-resume-course').attr('href').endswith('/sequential/' + sequential.url_name))
+        self.assertTrue(content('.action-resume-course').attr('href').endswith('/vertical/' + vertical.url_name))
+
+    def test_resume_course_with_completion_api(self):
+        """
+        Tests completion API resume button functionality
+
+        """
+        # Enable the waffle flag for all tests
+        self.override_waffle_switch(True)
+
+        # Course, Verticals
+        course = self.course
+        course_key = CourseKey.from_string(str(course.id))
+        vertical1 = course.children[0].children[0].children[0]
+        vertical2 = course.children[0].children[1].children[0]
+
+        # Fake visit to sequence1/vertical1
+        block_key = UsageKey.from_string(unicode(vertical1.location))
+        completion = 1.0
+        BlockCompletion.objects.submit_completion(
```

Comment:
Instead of calling `BlockCompletion.objects.submit_completion` function, what about making a fake ajax call to complete the block?

Reply:
@schenedx That seems like we're testing that layer of functionality rather than just testing the behavior of the resume button. These are intended to only test resume button behavior. Theoretically, the tests for page/completion exist elsewhere.

Reply:
OK.. I can buy that argument